src/MySql/custom/New-AzMySqlFlexibleServer.ps1

```diff
@@ -147,60 +183,418 @@ function New-AzMySqlFlexibleServer {
+                $NetworkParameters.VnetName = $NetworkParameters.Vnet
+                $NetworkParameters.SubnetName = 'Subnet' + $NetworkParameters.Name
+                $Subnet = CreateVnetSubnet $NetworkParameters
+                IsValidRgLocation $Subnet.Id $NetworkParameters 
+            }
+            Else {
+                Throw "Incorrectly formed Vnet id or Vnet name"
+            }
+        }
+        Else { # Both Vnet and Subnet provided
+            IF ($NetworkParameters.Vnet -Match $AZURE_ARMNAME -And $NetworkParameters.Subnet -Match $AZURE_ARMNAME) {
+                $NetworkParameters.VnetName = $NetworkParameters.Vnet
+                $NetworkParameters.SubnetName = $NetworkParameters.Subnet
+                $Subnet = CreateVnetSubnet $NetworkParameters
+            }
+            Else {
+                If ($NetworkParameters.ContainsKey('SubnetPrefix') -And $NetworkParameters.ContainsKey('VnetPrefix')) {
+                    $Msg = "If you pass an address prefix, please consider passing a name (instead of Id) for a subnet or vnet."
+                }
+                Else { $Msg = "If you pass both --vnet and --subnet, consider passing names instead of ids." }
+                Throw $Msg
+            }
+        }
+    }
+    # Handling create command without arguments
+    ElseIf (!$NetworkParameters.ContainsKey('PublicAccess') -And !$NetworkParameters.ContainsKey('Subnet') -And !$NetworkParameters.ContainsKey('Vnet')) {
+        $NetworkParameters.VnetName = 'VNET' + $NetworkParameters.Name
+        $NetworkParameters.SubnetName = 'Subnet' + $NetworkParameters.Name
+        $NetworkParameters.VnetPrefix = $DEFAULT_VNET_PREFIX
+        $NetworkParameters.SubnetPrefix = $DEFAULT_SUBNET_PREFIX
+
+        $Subnet = CreateVnetSubnet $NetworkParameters
+    }
+    return $NetworkParameters
+}
+
+function GetSubnetId($ResourceGroupName, $VnetName, $SubnetName){
+    If (!($ResourceGroupName -is [String])){ $ResourceGroupName = $ResourceGroupName[0]}
+    $Vnet = Get-AzVirtualNetwork -Name $VnetName -ResourceGroupName $ResourceGroupName
+    $Subnet = Get-AzVirtualNetworkSubnetConfig -Name $SubnetName -VirtualNetwork $Vnet
+    Return $Subnet.Id 
+}
+
+function CreateVnetSubnet($Parameters){
+    If (!$Parameters.ContainsKey('SubnetPrefix')){$Parameters.SubnetPrefix = $DEFAULT_SUBNET_PREFIX}
+    If (!$Parameters.ContainsKey('VnetPrefix')){$Parameters.VnetPrefix = $DEFAULT_VNET_PREFIX}
+
+    Try {
+        Get-AzVirtualNetwork -Name $Parameters.VnetName -ResourceGroupName $Parameters.ResourceGroupName -ErrorAction Stop
+        $Msg = "The provided vnet does exist."
+        Write-Host $Msg
+    }
+    Catch {
+        $Msg = "Creating new vnet {0} in resource group {1}" -f $Parameters.VnetName, $Parameters.ResourceGroupName
+        Write-Host $Msg
+        New-AzVirtualNetwork -Name $Parameters.VnetName -ResourceGroupName $Parameters.ResourceGroupName -Location $Parameters.Location -AddressPrefix $Parameters.VnetPrefix -Force
```

Comment:
Support should process for create virtual network

Reply:
Added!